server/src/main/java/org/elasticsearch/http/HttpRequestLineAndHeaders.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http;
+
+import org.elasticsearch.rest.RestRequest;
+
+import java.util.List;
+import java.util.Map;
+
+public interface HttpRequestLineAndHeaders {
+
+    /**
+     * Returns the HTTP method used in the HTTP request.
+     *
+     * @return the {@link RestRequest.Method} used in the REST request
+     * @throws IllegalArgumentException if the HTTP method is invalid
+     */
+    RestRequest.Method method();
```

Comment:
I would like to refactor `RestRequest.Method` and bring the `Method` inner class to this interface, but the changeset is going to be huge. Should I go ahead?

Reply:
If it is massive, I would prefer a standalone PR. If it is just moving stuff around, that is pretty quick/easy to review.